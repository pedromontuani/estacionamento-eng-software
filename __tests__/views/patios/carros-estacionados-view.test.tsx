import React from 'react';
import * as ReactNavigation from '@react-navigation/native';
import {rendererCreate} from '../../utils';

import ParkedCarsView from '../../../src/views/patios/carros-estacionados-view';

describe('Testing parked cars view', () => {
  beforeEach(() => {
    const useRoute = jest.spyOn(ReactNavigation, 'useRoute');
    // @ts-ignore
    useRoute.mockImplementationOnce(() => ({
      params: {patio: {id: 1, nome: 'Patio 1'}},
    }));
  });

  test('should match snapshot', async () => {
    const tree = rendererCreate(<ParkedCarsView />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
