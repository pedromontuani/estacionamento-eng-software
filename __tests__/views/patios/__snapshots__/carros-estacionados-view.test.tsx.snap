// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing patios view should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "flex": 1,
      "padding": 24,
    }
  }
>
  <Text>
    Carros estacionados em: 
    Patio 1
  </Text>
  <RCTScrollView
    ListEmptyComponent={
      <Text>
        Nenhum carro estacionado
      </Text>
    }
    contentContainerStyle={
      Object {
        "padding": 12,
      }
    }
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "marginHorizontal": -24,
      }
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <Text
        onLayout={[Function]}
      >
        Nenhum carro estacionado
      </Text>
    </View>
  </RCTScrollView>
</View>
`;
